[RFC] reexport libusb types from libusb1-sys

tanriol opened the pull request:
This change would make it possible to enumerate and select devices in
ways `libftdi` does not directly support, as well as interoperate with
`rusb` and/or other USB-related crates.
On the other hand, it may be possible to get really confusing errors if
`libusb1-sys` and `libftdi` get linked against two different `libusb`
versions.

@roqvist / @cr1901 Any opinions? Merge / close / under a separate feature only?

tanriol commented:
Also this would require bumping MSRV to 1.32... not much of a bump, but still.

tanriol committed:
update documentation for libusb1-sys

cr1901 commented:
@tanriol I am fine w/ this change and merging without a feature gate. It is semi-normal to use `libusb` features from `libftdi`. Though atm, I don't plan on providing any functions in `ftdi-rs` that use those types directly.

tanriol committed:
make placeholders for libusb1 types internal-only

They should not be converted to `libusb1-sys` types because they can
refer to different `libusb` versions (system vs bundled).

tanriol committed:
reexport libusb types from libusb1-sys

This change would make it possible to enumerate and select devices in
ways `libftdi` does not directly support, as well as interoperate with
`rusb` and/or other USB-related crates.
On the other hand, it may be possible to get really confusing errors if
`libusb1-sys` and `libftdi` get linked against two different `libusb`
versions. Detecting that correctly is not possible at the moment as the
current `libusb1-sys` versions do not export the required metadata.

tanriol committed:
build: abort compilation for static `libusb1-sys`

It would be incorrect to link dynamically to system `libftdi1` and
integrate with `libusb` that's built statically at the same time.

tanriol commented:
Note: I've updated this branch with a new approach. It is pending a `libusb1-sys` PR I'm going to make.

tanriol committed:
ci: use strategies to reduce boilerplate

tanriol committed:
ci: add tests for the `libusb1-sys` feature
